docs/react/guides/nextjs.md

```diff
@@ -142,3 +142,75 @@ export default Index;
 Without Nx, creating a new shared library can take from several hours or even weeks: a new repo needs to be provisioned, CI needs to be set up, etc.. In an Nx Workspace, it only takes minutes.
 
 You can share React components between multiple Next.js applications. You can also share web components between Next.js and plain React applications. You can even share code between the backend and the frontend. All can be done without any unnecessary ceremony.
+
+## Deploying to Vercel
+
+You may know that the company behind Next.js, Vercel, has a great hosting platform offering that is developed in tandem with Next.js itself to offer a great overall developer and user experience.
+
+In order to deploy your Next.js application from your Nx workspace you should do the following:
+
+### Verify the project's next.config.js
+
+Let's continue to use our `tuskdesk` example from above, and so we need to check out our config at `apps/tuskdesk/next.config.js`. If you created the application using a recent (at the time of writing) version of Nx, such as Nx 11, then you will likely see the following in that config by default:
+
+```js
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const withNx = require('@nrwl/next/plugins/with-nx');
+
+module.exports = withNx({});
+```
+
+If you have a config which looks like that, leveraging the `withNx()` config plugin, **no further action is needed** in your config.
+
+If, however, you created the application using an older version of Nx, you may just see an empty object:
+
+```js
+module.exports = {};
+```
+
+If this is the case, then you must do one of the following:
+
+[Option 1] We would naturally highly recommend upgrading to the latest Nx (for many reasons), and updating the next.config.js to match the first example which leverages the `withNx()` config plugin.
+
+[Option 2] If for some reason you cannot upgrade to a version of Nx which provides the `withNx()` config plugin, you can manually add a `target` property to your exported config with a value of `'experimental-serverless-trace'`.
+
+E.g.
+
+```js
+module.exports = {
+  target: 'experimental-serverless-trace',
+  // ...You can of course have other Next.js config options specified here too, but the "target" is critical for Vercel deployments...
+};
```

Comment:
You might also need this to be able to build pages that use the serveless api:

```js
module.exports = {
  target: 'serverless',
....
};
```

Without it build will fail as it won't create serverless functions.

Reply:
I created projects using this approach and serverless functions are created just fine using this target, it was the recommendation of Vercel themselves.